modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/internal/model/listener/PortalInstanceLifecycleListenerImpl.java

```diff
@@ -18,11 +18,19 @@
 import com.liferay.commerce.currency.model.CommerceCurrency;
 import com.liferay.commerce.currency.service.CommerceCurrencyLocalService;
 import com.liferay.portal.instance.lifecycle.BasePortalInstanceLifecycleListener;
+import com.liferay.portal.instance.lifecycle.PortalInstanceLifecycleListener;
 import com.liferay.portal.kernel.model.Company;
 
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
 /**
  * @author Marco Leo
  */
+@Component(
+	enabled = false, immediate = true,
+	service = PortalInstanceLifecycleListener.class
+)
 public class PortalInstanceLifecycleListenerImpl
```

Comment:
@JanisZhang When you make an OSGi component, you must ensure that it has a constructor without parameter. In this case, when you convert this class, since the `_commerceCurrencyLocalService` is now injected by the framework, you won't need that constructor, so just remove it.

Reply:
@dantewang I got it. Thanks!